doc/monitoring-stack-deployment/README.md

```diff
@@ -44,20 +54,124 @@ Then deploy prometheus server:
+```bash
+oc apply -f datasource-v4.yaml
+```
+
+6. Deploy grafana
+
+```bash
+oc apply -f grafana-v4.yaml
+```
+
+#### OpenShift versions from 4.14 and 3scale Operator version 2.15 upwards
+
+1. Install Grafana operator v5.12.0 from the Operator Hub.
+
+2. Create new service account for Prometheus
+
+```bash
+oc create serviceaccount prometheus-monitoring
+```
+
+3. Create a ClusterRoleBinding to give Prometheus ServiceAccount the RBAC permissions required to scrape metrics
+
+```bash
+cat << EOF | oc create -f -
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-monitoring
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-monitoring
+    namespace: <3scale namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+EOF
+```
+
+4. Create a token for the prometheus-monitoring ServiceAccount
+
+```bash
+oc create token prometheus-monitoring
+```
+
+Bare in mind that this token will expire which means the Prometheus will lose access to required resource. You can add the `--duration 0s` to token creation to have token active forever. Refer to official OpenShift documentation on managing token lifetime.
+
+5. Update the file `3scale-scrape-configs.yaml` bearer_token field with the token generated from the above.
+
+6. Create additional-scrape-config secret:
+
+```bash
+oc create secret generic additional-scrape-configs --from-file=3scale-scrape-configs.yaml=./3scale-scrape-configs.yaml
+```
+
+7. Deploy prometheus
+
+In `prometheus.yaml` file provided, fill the `spec.externalUrl` field with the external URL. The URL template should be:
+
+```yaml
+spec:
+  ...
+  externalUrl: https://prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
+```
+
+Then deploy prometheus server:
+
+```bash
+oc apply -f prometheus.yaml
+```
+
+8. Create Prometheus route
 
 ```bash
 oc expose service prometheus-operated --hostname prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
 ```
 
-7. Deploy grafana datasource
+9. Deploy grafana datasource
 
 ```bash
-oc apply -f datasource.yaml
+oc apply -f datasource-v5.yaml
 ```
 
-8. Deploy grafana
+10. Deploy grafana
 
 ```bash
-oc apply -f grafana.yaml
+oc apply -f grafana-v5.yaml
 ```
+
+#### Migration from V4 to V5
+
+1. Remove the Grafana Operator v4
+
+At this stage, Grafana Operator will be removed but the v4 dashboards CRDs will remain intact. 
+
+2. Install Grafana Operator v5
+
+At this stage, if 3scale operator reconciliation loop was triggered, operator will recongnize that v5 is present and will create v5 dashboards, but won't remove the v4 dashboards.
+
+3. Create Grafana v5 datasource (instructions above)
+4. Create Grafana v5 grafana (instructions above)
+5. Restart 3scale Operator
+
+Restart might be required since by design, 3scale operator does not watch over the grafana dashboards, but reconciles them as part of regular reconciliation loops.
+
+6. Delete the Grafana v4 custom resource
+7. Expose Grafana v5 route
```

Comment:
exposing not required if new grafana cr is applied after deletion of previous one